src/process.ts

```diff
@@ -92,15 +99,24 @@ export interface ProcessCallbackDecorator {
 /**
  * CreateProcess factory interface
  */
-export interface CreateProcess {
-	<T>(commands: (Command<T>[] | Command<T>)[], callback?: ProcessCallback<T>): Process<T>;
+export interface CreateProcess<T = any, P extends object = object> {
```

Comment:
Would it be worthwhile to provide overrides for 1 to N commands where the payload type is the intersection of the payload types for each command?